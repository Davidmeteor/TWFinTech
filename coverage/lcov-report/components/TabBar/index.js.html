<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/TabBar/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/TabBar</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>26/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>20/20</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">// https://github.com/skv-headless/react-native-scrollable-tab-view/blob/master/DefaultTabBar.js
'use strict'
import React, { PropTypes } from 'react'
import ScrollableTabView, { DefaultTabBar } from 'react-native-scrollable-tab-view'
import styles from './styles'
import MainStyle from '../../styles'
import { View, Text } from 'react-native'
&nbsp;
const TabBar = ({
  tabBarPosition,
  activeTextColor,
  underline,
  tabStyle,
  style,
  textStyle,
  onChangeTab,
  initialPage,
  children,
}) =&gt; (
  &lt;ScrollableTabView
    tabBarPosition = {tabBarPosition}
    onChangeTab={onChangeTab}
    initialPage={initialPage}
    renderTabBar={() =&gt;
      &lt;DefaultTabBar
        activeTextColor = {activeTextColor}
        underlineStyle = {underline}
        tabStyle = {tabStyle}
        style = {style}
        textStyle = {textStyle}
      /&gt;}
  &gt;
    {children}
  &lt;/ScrollableTabView&gt;
)
&nbsp;
TabBar.propTypes = {
  tabBarPosition: PropTypes.string,
  activeTextColor: PropTypes.string,
  underline: View.propTypes.style,
  tabStyle: View.propTypes.style,
  style: View.propTypes.style,
  textStyle: Text.propTypes.style,
  onChangeTab: PropTypes.func,
  initialPage: PropTypes.number,
}
TabBar.defaultProps = {
  tabBarPosition: 'top',
  activeTextColor: MainStyle.color.main,
  underline: styles.underline,
  tabStyle: styles.tabStyle,
  style: styles.tabBar,
  textStyle: styles.text,
  onChangeTab: () =&gt; <span class="fstat-no" title="function not covered" >{},</span>
  initialPage: 0,
}
&nbsp;
export default TabBar
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Sat Jan 07 2017 10:28:10 GMT-0800 (PST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
