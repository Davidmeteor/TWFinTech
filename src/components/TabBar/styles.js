import { StyleSheet } from 'react-native'
import MainStyle from '../../styles'

export default StyleSheet.create({
  tabStyle: {
    paddingTop: 7,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  tabBar: {
    height: 40,
    flexDirection: 'row',
    justifyContent: 'space-around',
    borderWidth: 1,
    borderTopWidth: 0,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    borderColor: MainStyle.color.main,
  },
  underline: {
    backgroundColor: MainStyle.color.main,
  },
  text: {
    fontSize: MainStyle.font.medium,
  },
})
