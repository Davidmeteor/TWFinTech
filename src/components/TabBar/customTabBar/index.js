'use strict';

import React, {Component} from 'react'
import { View, TouchableOpacity, Text } from 'react-native'
import styles from './styles'
import Icon from 'react-native-vector-icons/Ionicons'
import MainStyle from '../../../styles'

class CustomTabBar extends Component {

	static propTypes = {
		goToPage: React.PropTypes.func, // jump to the corresponding func of tab
		activeTab: React.PropTypes.number, // current selected tab
		tabs: React.PropTypes.array, // the set of all tab
		tabNames: React.PropTypes.array, // tab name
		tabIconNames: React.PropTypes.array, // tab icon
	}

	setAnimationValue({value}) {
		//console.log(value);
	}

	componentDidMount() {
		// Animated.Value range [0, # of tabs - 1]
		this.props.scrollValue.addListener(this.setAnimationValue);
	}

	renderTabOption(tab, i) {
		let color = this.props.activeTab == i ? MainStyle.color.main : "#ADADAD"; // check the # i, if it's the current selected one, change the color
		return (
            <TouchableOpacity key={this.props.tabIconNames[i]} onPress={()=>this.props.goToPage(i)} style={styles.tab}>
				<View style={styles.tabItem}>
					<Icon
						name={this.props.tabIconNames[i]} // icon
						size={30}
						color={color}/>
					<Text style={{color: color}}>
						{this.props.tabNames[i]}
					</Text>
				</View>
			</TouchableOpacity>
		);
	}

	render() {
		return (
			<View style={styles.tabs}>
				{this.props.tabs.map((tab, i) => this.renderTabOption(tab, i))}
			</View>
		);
	}
}

export default CustomTabBar;