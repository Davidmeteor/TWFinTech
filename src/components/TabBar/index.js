// https://github.com/skv-headless/react-native-scrollable-tab-view/blob/master/DefaultTabBar.js
'use strict'
import React, { PropTypes } from 'react'
import ScrollableTabView, { DefaultTabBar } from 'react-native-scrollable-tab-view'
import styles from './styles'
import MainStyle from '../../styles'
import { View, Text } from 'react-native'
import CustomTabBar from './customTabBar'

const TabBar = ({
  tabBarPosition,
  activeTextColor,
  underline,
  tabStyle,
  style,
  textStyle,
  onChangeTab,
  initialPage,
  tabNames,
  tabIconNames,
  children,
}) => (
  <ScrollableTabView
    tabBarPosition = {tabBarPosition}
    onChangeTab={onChangeTab}
    initialPage={initialPage}
    renderTabBar={() =>
      <CustomTabBar
        tabNames={tabNames} 
        tabIconNames={tabIconNames}
      />}
  >
    {children}
  </ScrollableTabView>
)

TabBar.propTypes = {
  tabBarPosition: PropTypes.string,
  activeTextColor: PropTypes.string,
  underline: View.propTypes.style,
  tabStyle: View.propTypes.style,
  style: View.propTypes.style,
  textStyle: Text.propTypes.style,
  onChangeTab: PropTypes.func,
  initialPage: PropTypes.number,
}
TabBar.defaultProps = {
  tabBarPosition: 'bottom',
  activeTextColor: MainStyle.color.main,
  underline: styles.underline,
  tabStyle: styles.tabStyle,
  style: styles.tabBar,
  textStyle: styles.text,
  onChangeTab: () => {},
  initialPage: 0,
}

export default TabBar
