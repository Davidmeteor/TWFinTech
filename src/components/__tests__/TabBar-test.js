/**
 * # TabBar-test.js
 *
 * This class tests that the tabBar displays correctly
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 */
'use strict'
import React from 'react'
import TabBar from '../TabBar'
import renderer from 'react-test-renderer'
import { View } from 'react-native'

describe('<TabBar />', () => {
  const innerView = {
    flex: 1,
  }

  it('render TabBar', () => {
    const view = (
      <TabBar>
        <View
          style={innerView}
          tabLabel="first page"
        />
        <View
          style={innerView}
          tabLabel="Second page"
        />
      </TabBar>)
    const tree = renderer.create(view).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render TabBar with initial page', () => {
    const props = {
      initialPage: 0,
    }

    const view = (
      <TabBar {...props}>
        <View
          style={innerView}
          tabLabel="First page"
        />
        <View
          style={innerView}
          tabLabel="Second page"
        />
      </TabBar>)
    const tree = renderer.create(view).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
