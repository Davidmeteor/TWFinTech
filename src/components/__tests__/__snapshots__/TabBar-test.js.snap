exports[`<TabBar /> render TabBar 1`] = `
<View
  onLayout={[Function bound _handleLayout]}
  style={
    Array [
      Object {
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around"
        },
        Object {
          "backgroundColor": null
        },
        Object {
          "borderColor": "#F78B6D",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around"
        }
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="first page"
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "flex": 1,
          "opacity": 1
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 10
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 7
            }
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#F78B6D",
                "fontWeight": "bold"
              },
              Object {
                "fontSize": 16
              }
            ]
          }>
          first page
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="Second page"
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "flex": 1,
          "opacity": 1
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 10
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 7
            }
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "normal"
              },
              Object {
                "fontSize": 16
              }
            ]
          }>
          Second page
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F78B6D",
          "bottom": 0,
          "height": 4,
          "left": 0,
          "position": "absolute",
          "width": 187.5
        }
      } />
  </View>
  <ScrollView
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    contentOffset={
      Object {
        "x": 0
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    onMomentumScrollBegin={[Function bound _onMomentumScrollBeginAndEnd]}
    onMomentumScrollEnd={[Function bound _onMomentumScrollBeginAndEnd]}
    onScroll={[Function onScroll]}
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={16}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}>
    <View
      style={
        Object {
          "width": 375
        }
      }>
      <View
        style={
          Object {
            "flex": 1
          }
        }
        tabLabel="first page" />
    </View>
    <View
      style={
        Object {
          "width": 375
        }
      }>
      <View
        tabLabel="Second page" />
    </View>
  </ScrollView>
</View>
`;

exports[`<TabBar /> render TabBar with initial page 1`] = `
<View
  onLayout={[Function bound _handleLayout]}
  style={
    Array [
      Object {
        "flex": 1
      },
      undefined
    ]
  }>
  <View
    style={
      Array [
        Object {
          "borderColor": "#ccc",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "space-around"
        },
        Object {
          "backgroundColor": null
        },
        Object {
          "borderColor": "#F78B6D",
          "borderLeftWidth": 0,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 40,
          "justifyContent": "space-around"
        }
      ]
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="First page"
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "flex": 1,
          "opacity": 1
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 10
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 7
            }
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#F78B6D",
                "fontWeight": "bold"
              },
              Object {
                "fontSize": 16
              }
            ]
          }>
          First page
        </Text>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel="Second page"
      accessibilityTraits="button"
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function bound touchableHandleResponderGrant]}
      onResponderMove={[Function bound touchableHandleResponderMove]}
      onResponderRelease={[Function bound touchableHandleResponderRelease]}
      onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
      onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
      onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
      style={
        Object {
          "flex": 1,
          "opacity": 1
        }
      }
      testID={undefined}>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingBottom": 10
            },
            Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingTop": 7
            }
          ]
        }>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "black",
                "fontWeight": "normal"
              },
              Object {
                "fontSize": 16
              }
            ]
          }>
          Second page
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#F78B6D",
          "bottom": 0,
          "height": 4,
          "left": 0,
          "position": "absolute",
          "width": 187.5
        }
      } />
  </View>
  <ScrollView
    alwaysBounceVertical={false}
    automaticallyAdjustContentInsets={false}
    contentOffset={
      Object {
        "x": 0
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    onMomentumScrollBegin={[Function bound _onMomentumScrollBeginAndEnd]}
    onMomentumScrollEnd={[Function bound _onMomentumScrollBeginAndEnd]}
    onScroll={[Function onScroll]}
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={16}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}>
    <View
      style={
        Object {
          "width": 375
        }
      }>
      <View
        style={
          Object {
            "flex": 1
          }
        }
        tabLabel="First page" />
    </View>
    <View
      style={
        Object {
          "width": 375
        }
      }>
      <View
        tabLabel="Second page" />
    </View>
  </ScrollView>
</View>
`;
